### ISSUE TITLE
night mode

### ISSUE THREAD

**Issue by karussell** [graphhopper/graphhopper-navigation-example-legacy#37]
This would be cool at some point

**Comment by boldtrn** [comment]
In theory the SDK can handle different map styles for day and night mode and I think an automatic switch depending on the time is also possible.

Right now, we don't have a night map style though.

**Comment by karussell** [comment]
:+1:

**Comment by boldtrn** [comment]
I added the dark-matter map style to [Mapilion](https://mapilion.com/) - you can try it by going to Mapilion, click on `Dark Matter` next to the hillshading options. I could set this as night style for navigation? I don't think dark-matter is ideal for navigating, because it misses many essential details, but it would be good enough to show the switch between day/night.

**Comment by karussell** [comment]
Hmmh, the contrast is too weak. But this is also not such a high prio :)

**Comment by karussell** [comment]
When using minimum light the current theme also works okayish

**Comment by boldtrn** [comment]
It's up to you. I think it should be straight forward to change the night mode setting. I also think the Dark-Matter style is not ideal, but it could be better than our current style and show users that they can set two different styles. We could also try and make this configurable in the settings (but it might not work during navigation then, because we set the navigation values in xml and I am not sure if we could do this in the regular code).

**Comment by karussell** [comment]
Let's wait for real demand as this seems to be not that easy - we would need a light sensor access as this should be enabled also in a tunnel IMO.

**Comment by boldtrn** [comment]
> we would need a light sensor access as this should be enabled also in a tunnel IMO

Yes, but I think the better way to approach this is something like PathDetails than using a light sensor. This has been added like this to the Mapbox SDK [here](https://github.com/mapbox/mapbox-navigation-android/pull/1392).
